src/legacy/core_plugins/data/public/index.ts

```diff
@@ -25,25 +34,34 @@ class DataPlugin {
   private readonly indexPatterns: IndexPatternsService;
   private readonly search: SearchService;
   private readonly query: QueryService;
+  private readonly expressionExecutor: ExpressionExecutorService;
 
   constructor() {
     this.indexPatterns = new IndexPatternsService();
     this.query = new QueryService();
     this.search = new SearchService();
+    this.expressionExecutor = new ExpressionExecutorService();
   }
 
   public setup() {
     return {
       indexPatterns: this.indexPatterns.setup(),
       search: this.search.setup(),
       query: this.query.setup(),
+      expressionExecutor: this.expressionExecutor.setup(null, {
```

Comment:
lets name it interpreter service

Reply:
Sounds good - this means we will merge it with the current interpreter plugin, right?

Reply:
++
`data.expressions.fromAst()` and `data.expression.toAst()` probably make more sense then
`data.expressions.run()` is also fine and `data.expressions.ExpressionRenderer` component
it would be nice if we could have `data.expressions.interpreter` instead of `data.expressions.getInterpreter()`
and then the registries:
`data.expressions.functionRegistry`, `data.expressions.renderersRegistry`, `data.expressions.typesRegistry`